crates/codegen/language/definition/src/model/utils/mod.rs

```diff
@@ -0,0 +1,3 @@
+mod identifier;
+
+pub use identifier::*;
```

Comment:
This is `model::utils` and I'm not really advocating for not re-exporting items at the same module, which is very useful.

I'm advocating for not defaulting to pub-glob-reexporting everywhere, unless there's a benefit in doing so.

In this case, this should just be:
```rust
pub use identifier::Identifier;
pub use version_specifier::VersionSpecifier;
```

Because this, at a glance, tells the reader what types are exported directly from this module without forcing them into reading the child modules and seeing what's exported by default.

This rings even more true for modules that re-export only single items from their child modules. One has to go into `mod separated` to discover that only a single `SeparatedItem` is exported.

Instead of 
```rust
pub use fragment::*;
pub use keyword::*;
pub use scanner::*;
pub use token::*;
pub use trivia::*;
```
a better story is being told to the user with
```rust
pub use fragment::FragmentItem;
pub use keyword::{KeywordItem, KeywordDefinition, KeywordValue};
pub use scanner::Scanner;
pub use token::{TokenItem, TokenDefinition};
pub use trivia::{TriviaItem, TriviaParser};
```
The reader can see the `Item`-suffix pattern, understand that some of these use supplementary `Definition`s (which uses `Value` in one case) and in general helps build a better intuition without forcing the user to do the file scanning themselves or run a doc-gen tool in order to build a better mental model.

Pub-glob-use has its places like re-exporting stuff from
- inline modules (introducing them may help with privacy/scope but we want to use them at the parent/current module level),
- generated ones (where we have no control/knowledge of expanded items)
- or for other reasons (e.g. OS-specific submodules in [`rustc`](https://github.com/rust-lang/rust/blob/caf730043232affb6b10d1393895998cb4968520/library/std/src/sys/mod.rs#L31))

but defaulting to it everywhere makes it so that it's easier to type and harder to understand, whereas the opposite should be true, instead.

Reply:
Fully agree, although a bit more work to maintain/keep up to date manually.I wish RA introduced better commands for that :(